packages/lib-react-components/src/Tab/Tab.spec.js

```diff
@@ -1,42 +1,72 @@
-import zooTheme from '@zooniverse/grommet-theme';
 import React from 'react'
-import { shallow, mount } from 'enzyme'
+import { render, screen, within } from '@testing-library/react'
+import zooTheme from '@zooniverse/grommet-theme'
 import { Grommet } from 'grommet'
+import { Key } from 'grommet-icons'
 
 import Tabs from '../Tabs'
 import Tab from './Tab'
 import SpacedText from '../SpacedText'
 
-describe('<Tab />', function () {
-  it('should render without crashing', function () {
-    const wrapper = shallow(<Tab title='Foobar' />)
-    expect(wrapper).to.be.ok()
-  })
+describe('Component > Tab', function () {
+  describe('with standard props', function () {
+    beforeEach(function () {
+      render(
+        <Grommet theme={zooTheme} >
+          <Tabs>
+            <Tab title='apples'>An apple is a red fruit.</Tab>
+            <Tab title='bananas'>A banana is a long fruit.</Tab>
+            <Tab title='cherries'>A cherry is a stone fruit.</Tab>
+          </Tabs>
+        </Grommet>
+      )
+    })
+
+    it('should render without crashing', function () {
+      const tabsContainer = screen.getByRole('tablist')
+      const arrayOfTabs = screen.getAllByRole('tab')
+      const tabPanel = screen.getByRole('tabpanel')
+      expect(tabsContainer).to.exist()
+      expect(arrayOfTabs).to.have.length(3)
+      expect(tabPanel).to.exist()
+    })
+
+    it('should render the `title` prop as standard text if it\'s a string', function () {
+      const arrayOfTabs = screen.getAllByRole('tab')
 
-  it('should render the `title` prop using the SpacedText component if it\'s a string', function () {
-    const TITLE = 'Foobar'
-    const wrapper = mount(
-      <Grommet theme={zooTheme}>
-        <Tabs>
-          <Tab title={TITLE} />
-        </Tabs>
-      </Grommet>
-    )
-    const tabTitle = wrapper.find(SpacedText)
-    expect(tabTitle).to.have.lengthOf(1)
-    expect(tabTitle.text()).to.equal(TITLE)
+      const tab0 = within(arrayOfTabs[0])
+      const tab1 = within(arrayOfTabs[1])
+      const tab2 = within(arrayOfTabs[2])
+
+      expect(tab0.getByText('apples')).to.exist()
+      expect(tab1.getByText('bananas')).to.exist()
+      expect(tab2.getByText('cherries')).to.exist()
+    })
   })
 
-  it('should should pass through the `title` prop if it\'s a component', function () {
-    const TITLE = () => (<div>Foobar</div>)
-    const wrapper = mount(
-      <Grommet theme={zooTheme}>
-        <Tabs>
-          <Tab title={<TITLE />} />
-        </Tabs>
-      </Grommet>
-    )
-    const tabTitle = wrapper.find(TITLE)
-    expect(tabTitle).to.have.lengthOf(1)
+  describe('with complex props', function () {
+    it('should should pass through the `title` prop if it\'s a component', function () {
+      const ComplexTitle = () => (
+        <div>
+          <Key size='small' />
+          &nbsp;
+          <b data-testid="b-tag">Special</b>
+        </div>
```

Comment:
```suggestion
        <span>
          <Key size='small' />
          &nbsp;
          <b data-testid="b-tag">Special</b>
        </span>
```

I suspect you want the tab title to render inline, in which case it should use a text `<span>` rather than `<div>`.

Also, I suspect that the text label for this tab is 'Key Special' but that you intend the tab name to be 'Special'. You can add a test to get the tab by name and check that.

Reply:
OOI, if any HTML markup is allowed inside tab titles, what happens if you pass in an ordered or unordered list? We should probably restrict the content to `span` and other inline elements, like the content model for `<legend>` or `<label>`.

Reply:
Grommet tabs are implemented as `<button>` tags. `div`, `ul`, `p`, headings etc. are invalid content for HTML buttons.

Reply:
`<span>` makes sense. Making this change now.